eng/Version.Details.xml

```diff
@@ -8,6 +8,21 @@
       <Uri>https://github.com/dotnet/runtime</Uri>
       <Sha>6f9d6569684cc17015aa6fc5f9d9a5f7580ade97</Sha>
     </Dependency>
+    <!-- Necessary for source-build. This allows the live version of the package to be used by source-build. -->
+    <Dependency Name="Microsoft.Build.Framework" Version="17.3.2">
+      <Uri>https://github.com/dotnet/msbuild</Uri>
+      <Sha>561848881bab01749e6d8b03be2869a18ca944f7</Sha>
+    </Dependency>
+    <!-- Necessary for source-build. This allows the live version of the package to be used by source-build. -->
+    <Dependency Name="Microsoft.Build.Tasks.Core" Version="17.3.2">
+      <Uri>https://github.com/dotnet/msbuild</Uri>
+      <Sha>561848881bab01749e6d8b03be2869a18ca944f7</Sha>
+    </Dependency>
```

Comment:
We should not be adding dependencies that have no corresponding versions.props entries.  There is a degree of flexibility permitted in sharing the versions.props for like dependencies as this repo is doing.

Reply:
Would it be appropriate to define and use `MicrosoftBuildFrameworkVersion` and `MicrosoftBuildTasksCoreVersion` for the packages, respectively?

Reply:
> Would it be appropriate to define and use `MicrosoftBuildFrameworkVersion` and `MicrosoftBuildTasksCoreVersion` for the packages, respectively?

No, the repo is free to use this pattern of sharing the `MicrosoftBuildVersion` property for other packages. So we should maintain that pattern and adjust the Version.Details for only what is needed (in this case, just the `Microsoft.Build` dependency).

Reply:
Only adding the `Microsoft.Build` dependency did not fix all of the poison leaks; adding only that dependency fixed `System.Configuration.ConfigurationManager.dll`, `System.Reflection.MetadataLoadContext.dll`, and `System.Reflection.MetadataLoadContext.dll`. Adding `Microsoft.Build.Tasks.Core` to Version.Details and adding a package reference to it fixed all other remaining leaks.

I've updated this PR (https://github.com/dotnet/format/pull/2063/commits/c7009ef1a33aac2e3bb63f63ed59038235ee34aa) to remove `Microsoft.Build.Framework` from Version.Details and define/use `MicrosoftBuildTasksCoreVersion`.

Reply:
I'd like to better understand why that is. Can you share the two binlogs that compare these differences?

Reply:
Sure! 
[format_MsBuild_only.zip](https://github.com/dotnet/format/files/14015877/format_MsBuild_only.zip)
[format_MsBuild_and_MsBuildTasksCore.zip](https://github.com/dotnet/format/files/14015879/format_MsBuild_and_MsBuildTasksCore.zip)

Reply:
Ah, it's due to some incoherency. We're getting Microsoft.Build.Tasks.Core.17.3.2. It's Roslyn that's referencing an older version of MSBuild. Look at the project.assets.json snippet for the format project:

```xml
      "Microsoft.CodeAnalysis.Workspaces.MSBuild/4.10.0-1.24061.4": {
        "type": "package",
        "dependencies": {
          "Microsoft.Build": "17.3.2",
          "Microsoft.Build.Framework": "17.3.2",
          "Microsoft.Build.Tasks.Core": "17.3.2",
          "Microsoft.CodeAnalysis.Common": "[4.10.0-1.24061.4]",
          "Microsoft.CodeAnalysis.Workspaces.Common": "[4.10.0-1.24061.4]",
          "Microsoft.Extensions.Logging.Abstractions": "9.0.0-alpha.1.24061.26",
          "Newtonsoft.Json": "13.0.3",
          "System.Text.Json": "9.0.0-alpha.1.24061.26"
        },
```

We're flowing in the live Roslyn version, but that points to an older version of MSBuild. Since we didn't specify `PackageVersion` for `Microsoft.Build.Tasks.Core`, we were getting the 17.3.2 version. That's why your change to include that `PackageVersion` resolves the issue.